lib/cocoapods-core/specification.rb

```diff
@@ -43,8 +43,8 @@ def initialize(parent = nil, name = nil, test_specification = false)
       @consumers = {}
       @parent = parent
       @hash_value = nil
-      @test_specification = test_specification
       attributes_hash['name'] = name
+      attributes_hash['test_specification'] = test_specification
```

Comment:
Whether a specification is a test specification or not needs to actually be stored to the hash so any podspecs written to disk can be deserialized normally without losing this information.